
syntax = "proto2";



	message targetPowerStateParams{

		required bool targetPowerState = 2;
		
	}



	enum _controlPolicy { //Config mode representation
		
		AUTO_LUX = 0;
		AUTO_TIMER = 1;
		MANUAL = 2;

	}

	message targetControlPolicy{

		required _controlPolicy controlPolicy = 2;
		
	}

	message targetManualControlParams{

		optional uint32 targetBrightnessLevel = 3;
	
	}


	message targetAutoTimerParams{

		optional uint64 targetOnTime = 4;
		optional uint64 targetOffTime = 5;
	

	}
	
	message targetAutoLuxParams{

		optional uint32 targetOnLux = 6;
		optional uint32 targetOffLux = 7;

	}
	


	message targetConfigurations{

		required targetPowerStateParams  PowerState = 1;
		required targetControlPolicy ControlPolicy = 2;
		optional targetAutoTimerParams AutoTimerParams = 3;
		optional targetAutoLuxParams AutoLuxParams = 4;
	
	}	



